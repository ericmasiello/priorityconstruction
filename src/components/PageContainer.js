import React from 'react';
import styled from 'styled-components';
import { pxToRem } from '../styles/utils';
import { MAX_CONTENT_WIDTH } from '../styles/vars';
import { TagPropType } from '../propTypes';

export const PageContainer = ({ tag: Tag, ...rest }) => <Tag {...rest} />;

PageContainer.propTypes = {
  tag: TagPropType,
};

PageContainer.defaultProps = {
  tag: 'div',
};

PageContainer.displayName = 'PageContainer';

export default styled(PageContainer)`
  margin: 0 auto;
  max-width: ${pxToRem(MAX_CONTENT_WIDTH)};
  padding: 0px 1.0875rem 1.45rem;
`;
