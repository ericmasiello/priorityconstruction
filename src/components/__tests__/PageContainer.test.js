import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import PageContainer, { PageContainer as Raw } from '../PageContainer';

it('should render', () => {
  const component = renderer.create(<PageContainer />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
  const wrapper = mount(<Raw>Test</Raw>);

  expect(wrapper.text()).toBe('Test');
});

it('should render additional props', () => {
  const wrapper = shallow(<Raw data-test="my-attribute" />);

  expect(wrapper.find('[data-test="my-attribute"]')).toHaveLength(1);
});

it('should display additional classes', () => {
  const wrapper = shallow(<Raw className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render as a <div> by default', () => {
  const wrapper = shallow(<Raw />);

  expect(wrapper.type()).toBe('div');
});

it('should render with a custom tag', () => {
  const wrapper = shallow(<Raw tag="span" />);

  expect(wrapper.type()).toBe('span');
});
