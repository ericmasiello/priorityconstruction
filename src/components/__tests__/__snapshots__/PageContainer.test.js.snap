// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 64rem;
  padding: 0px 1.0875rem 1.45rem;
}

<div
  className="c0"
/>
`;
